src/inc/til/color.h

```diff
@@ -134,6 +134,19 @@ namespace til // Terminal Implementation Library. Also: "Today I Learned"
             };
         }
 
+        // Do a simple linear blend with `other`, taking a% of us and (1-a)% of
+        // other. The result will retain our original alpha channel value
+        constexpr color blend_with(const color& other)
+        {
+            auto result = *this;
+            const float a1 = a / 255.0f;
+            const float a2 = 1.0f - a1;
+            result.r = static_cast<uint8_t>((a1 * r) + (a2 * other.r));
+            result.g = static_cast<uint8_t>((a1 * g) + (a2 * other.g));
+            result.b = static_cast<uint8_t>((a1 * b) + (a2 * other.b));
+            return result;
+        }

```

Comment:
What type of blend operation is this? It's not the regular "source-over" alpha composition in straight alpha. That would be:
```suggestion
        // source-over alpha blending/composition.
        // `this` (source/top) will be blended "over" `destination` (bottom).
        // `this` and `destination` are expected to be in straight alpha.
        constexpr color blend_with(const color& destination)
        {
            const auto aa = a / 255.0f;
            const auto ab = destination.a / 255.0f;
            const auto aai = 1.0f - aa;

            const auto ra = a + destination.a * aai;
            const auto rr = (r * aa + other.r * ab * aai) / ra;
            const auto rg = (g * aa + other.g * ab * aai) / ra;
            const auto rb = (b * aa + other.b * ab * aai) / ra;

            return {
                static_cast<uint8_t>(rr + 0.5f),
                static_cast<uint8_t>(rg + 0.5f),
                static_cast<uint8_t>(rb + 0.5f),
                static_cast<uint8_t>(ra + 0.5f),
            };
        }
```

(https://en.wikipedia.org/wiki/Alpha_compositing#Description)

Apart from that I'm a bit surprised that `this` isn't treated as the destination but rather as the source color which is layered on top of the parameter. I feel like it'd be easier to understand if the order were swapped.

Reply:
There's also the problem that you technically have to convert any colors from the gamma-corrected color space into the linear color space first, the same way that `ColorHelper::GetLuminance` does it (via `pow`). But right now we treat VT colors as linear ones and UI colors as gamma-corrected ones, so you'd have to apply this correction depending on what the source colors are and for what they're used - if we wanted to do this properly it'd be a real hassle...